dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdMappingFileUploadTask.kt

```diff
@@ -179,5 +193,61 @@ open class DdMappingFileUploadTask
         return true
     }
 
+    private fun shrinkMappingFile(
+        mappingFile: File
+    ): File {
+        if (!mappingFileTrimIndents && mappingFilePackagesAliases.isEmpty()) return mappingFile
+
+        val shrinkedFile = File(mappingFile.parent, "mapping-shrinked.txt")
+        // sort is needed to have predictable replacement in the following case:
+        // imagine there are 2 keys - "androidx.work" and "androidx.work.Job", and the latter
+        // occurs much often than the rest under "androidx.work.*". So for the more efficient
+        // compression we need first to process the replacement of "androidx.work.Job" and only
+        // after that any possible prefix (which has a smaller length).
+        val replacements = mappingFilePackagesAliases.entries
+            .sortedByDescending { it.key.length }
+            .map {
+                Regex("${it.key}(?=\\W)") to it.value
+            }
+
+        mappingFile.readLines()
+            .map {
+                if (mappingFileTrimIndents) it.trimStart() else it
+            }
+            .map {
+                applyShortAliases(it, replacements)
+            }
+            .forEach {
+                shrinkedFile.appendText(it + "\n")
+            }
+
+        return shrinkedFile
+    }
+
+    private fun applyShortAliases(
+        line: String,
+        replacements: List<Pair<Regex, String>>
+    ): String {
+        val isLineWithRename = !line.startsWith(MAPPING_FILE_COMMENT_CHAR) &&
+            line.contains(MAPPING_FILE_CHANGE_DELIMITER)
+
+        return if (isLineWithRename) {
+            val (source, obfuscated) = line.split(MAPPING_FILE_CHANGE_DELIMITER)
```

Comment:
shouldnt we check on the number of `MAPPING_FILE_CHANGE_DELIMITER` in the string (if > 1?)

Reply:
or maybe use regex with group matching ?

Reply:
Seems it is not possible to have multiple `->` on the line with the current format, but indeed, I will add a check just in case.